src/migrations/20230408000001-rewrite_to_use_current_topics.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable max-len */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const loggedUser = '0';
+      const sessionSig = __filename;
+      const auditDescriptor = 'RUN MIGRATIONS';
+      await queryInterface.sequelize.query(
+        `SELECT
+                set_config('audit.loggedUser', '${loggedUser}', TRUE) as "loggedUser",
+                set_config('audit.transactionId', NULL, TRUE) as "transactionId",
+                set_config('audit.sessionSig', '${sessionSig}', TRUE) as "sessionSig",
+                set_config('audit.auditDescriptor', '${auditDescriptor}', TRUE) as "auditDescriptor";`,
+        { transaction },
+      );
+      // Update all the existing link tables to use the current topics,
+      // then update ActivityReports.topics
+      await queryInterface.sequelize.query(
+        `
+        CREATE TEMP TABLE renamed_topics AS
+        SELECT DISTINCT
+          lt.id legacy_tid,
+          lt."mapsTo" current_tid,
+          lt.name legacy_name,
+          ct.name current_name
+        FROM "Topics" lt
+        JOIN "Topics" ct
+          ON lt."mapsTo" = ct.id
+        WHERE lt.name IN ('Teaching Practices / Teacher-Child Interactions', 'Child Assessment, Development, Screening')
```

Comment:
After all topics are updated to the correct current_names will there be a followup to remove legacy topic names?

Reply:
My plan was to leave the historical topic names in the Topics table, which is the approach we've taken in the past.

Reply:
We dont need to do this, we do the mapping live when stats are collected